tests/phpunit/CRM/Utils/FileTest.php

```diff
@@ -73,4 +73,25 @@ public function testDuplicate($ext) {
     unlink($newFile);
   }
 
+  public function fileNames() {
+    $cases = [];
+    $cases[] = ['helloworld.txt', TRUE];
+    $cases[] = ['../helloworld.txt', FALSE];
+    // Test case seems to be failing for a strange reason
+    // $cases[] = ['\helloworld.txt', FALSE];
+    $cases[] = ['.helloworld', FALSE];
+    $cases[] = ['smartwatch_1736683_1280_9af3657015e8660cc234eb1601da871.jpg', TRUE];
+    return $cases;
+  }
```

Comment:
Oooh, dataProvider-based unit test! 👍

Reply:
now now @totten you aren't the ONLY one who writes those...